modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -418,7 +418,9 @@ public class TableManager implements IgniteTablesInternal, IgniteComponent {
 
     private final PartitionReplicaLifecycleManager partitionReplicaLifecycleManager;
 
-    private long implicitTransactionTimeout;
+    private long roTransactionTimeout;
```

Comment:
Tx timeout is tx specific attribute, meaning that tx1 will have 100ms as timeout and tx2 that is running at the same time 2000ms. In that case why `roTransactionTimeout` and `rwTransactionTimeout` are here?
Besides that, it's just an encapsulation leak - simply TableManager should not know about such things.
I'd rather say that on tx creation we should populate TransacionOptions with default timeout from cfg if it's not specified by the user and within tx flow use timeout from the tx context - no need to propagate txTimeout to InternalTableImpl etc.

Reply:
Do you agree that `implicitTransactionTimeout` should be removed with this PR? If yes, then it is clear why these options are there. If we should keep `implicitTransactionTimeout ` then, probably, I should rollback the change. 

I suppose we should remove `implicitTransactionTimeout` and refactor `TableManager` as you suggested. Am I right?

Reply:
+1 to remove implicitTransactionTimeout

Reply:
>In that case why roTransactionTimeout and rwTransactionTimeout are here?
I thought these properties define *default* TX timeouts. A default is global and not specific to a transaction.

Reply:
> I suppose we should remove implicitTransactionTimeout and refactor TableManager as you suggested. Am I right?

Yes.